glue_exp/tools/dendrogram_selection/dendrogram_selection.py

```diff
@@ -0,0 +1,134 @@
+import os
+import numpy as np
+
+from glue.viewers.common.qt.mouse_mode import MouseMode
+from glue.external.qt import QtGui
+
+from glue.core.edit_subset_mode import EditSubsetMode
+from glue.core.subset import MaskSubsetState
+
+__all__ = ['DendrogramSelectionTool']
+
+ROOT = os.path.dirname(__file__)
+
+WARN_THRESH = 10000000  # warn when dendrogram selection on large images
+
+
+class DendrogramSelectionTool(object):
+
+    def __init__(self, widget=None):
+        self.widget = widget
+        self.data_object = None
+
+    def _get_modes(self, axes):
+        self._mode = DendrogramMode(axes,
+                                   move_callback=self._dendrogram_roi,
+                                   release_callback=self._dendrogram_roi)
+        return [self._mode]
+
+    # @set_cursor(Qt.WaitCursor)
+    def _dendrogram_roi(self, mode):
+        """
+        Callback for DendrogramMode.
+        """
+        if mode._start_event is None or mode._end_event is None:
+            return
+
+        # TODO: how to get data from other widges??
+        data = self.widget.client.display_data
+        att = self.widget.client.display_attribute
+        dendro_att = 'structure'
+
+        print('data and att', data, att)
+
+        # assume I got the dendrogram structure here as d
+        d = data[dendro_att]  # here is ndarray data
+
```

Comment:
@astrofrog How to get dendrogram structure within `ImageWidget`? data['structure' will return a `ndarray` not a real dendrogram object.